components/dialog/stories/dialog.stories.js

```diff
@@ -46,39 +114,175 @@ export default {
+				category: "Content",
+			},
+			control: { type: "file", accept: ".svg,.png,.jpg,.jpeg,.webc" },
+			if: { arg: "layout", eq: "default" },
+		},
 	},
 	args: {
 		rootClass: "spectrum-Dialog",
-		isDismissable: true,
-		showModal: false,
+		hasFooter: true,
+		footer: "Do not show this message again.",
+		isDismissible: false,
 		isOpen: true,
+		showModal: false,
+		size: "medium",
+		layout: "default",
+		hasCheckbox: true,
 	},
 	parameters: {
 		actions: {
-			handles: ["click .spectrum-Dialog button"],
+			handles: [],
 		},
 		docs: {
 			story: {
+				inline: false,
 				height: "500px",
 			},
 		},
+		componentVersion: version,
 		status: {
 			type: "migrated",
 		},
 	},
 };
 
+const ExampleContent = "Standard dialog description. This should briefly communicate any additional information or context about the standard dialog title, to help users make one of the decisions offered by the buttons. Make it no more than a few short sentences.";
+
+/**
+ * The default size for dialog is medium.
+ */
 export const Default = Template.bind({});
 Default.args = {
 	heading: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
+	header: "*Required",
 	showModal: true,
 	content: [
 		() => Typography({
 			semantics: "body",
 			size: "m",
-			content: [
-				"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Auctor augue mauris augue neque gravida. Libero volutpat sed ornare arcu. Quisque egestas diam in arcu cursus euismod quis viverra. Posuere ac ut consequat semper viverra nam libero justo laoreet. Enim ut tellus elementum sagittis vitae et leo duis ut. Neque laoreet suspendisse interdum consectetur libero id faucibus nisl. Diam volutpat commodo sed egestas egestas. Dolor magna eget est lorem ipsum dolor. Vitae suscipit tellus mauris a diam maecenas sed. Turpis in eu mi bibendum neque egestas congue. Rhoncus est pellentesque elit ullamcorper dignissim cras lobortis."
-			]
+			content: ExampleContent,
 		}),
 	],
 };
+
+/* TODO: For all dialog stories: the "is-hidden-story" tags replicates "!dev" in older versions
+of Storybook, to remove the stories from the side navigation, reflecting the intended behavior.
+Remove "is-hidden-story" in favor of only "!dev" tags when possible.
+*/
+// ********* DOCS ONLY ********* //
+export const DefaultSmall = Template.bind({});
+DefaultSmall.tags = ["is-hidden-story", "!dev"];
+DefaultSmall.storyName = "Dialog - small",
+DefaultSmall.parameters = {
+	chromatic: { disableSnapshot: true },
+};
+DefaultSmall.args = {
+	...Default.args,
+	size: "small",
+};
+
+export const DefaultLarge = Template.bind({});
+DefaultLarge.tags = ["is-hidden-story", "!dev"];
+DefaultLarge.storyName = "Dialog - large",
+DefaultLarge.parameters = {
+	chromatic: { disableSnapshot: true },
+};
+DefaultLarge.args = {
+	...Default.args,
+	size: "large",
+};
+
+/**
+ * A dialog that can be dismissed without taking an action.
```

Comment:
For S2, dialogs display the close button and the button group. I removed part of the documentation from S1 here.